diff -Naur vokoscreenNG-3.0.4-orig/ vokoscreenNG-3.0.4/
diff -Naur vokoscreenNG-3.0.4-orig/src/region/QvkRegionButtonArrow.h vokoscreenNG-3.0.4/src/region/QvkRegionButtonArrow.h
--- vokoscreenNG-3.0.4-orig/src/region/QvkRegionButtonArrow.h   2020-06-29 10:25:08.557087808 -0400
+++ vokoscreenNG-3.0.4/src/region/QvkRegionButtonArrow.h        2020-06-29 10:27:18.634550625 -0400
@@ -24,6 +24,7 @@
 #define REGIONBUTTONARROW_H

 #include <QPainter>
+#include <QPainterPath>

 class QvkRegionButtonArrow : public QObject
 {
