diff --unified --recursive --text anbox/CMakeLists.txt anbox.new/CMakeLists.txt
--- anbox/CMakeLists.txt	2019-11-01 00:30:23.841963699 -0400
+++ anbox.new/CMakeLists.txt	2019-11-01 00:33:08.626894747 -0400
@@ -94,9 +94,7 @@
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DEGL_NO_X11")
 
-if((Protobuf_VERSION GREATER "3.7") OR (Protobuf_VERSION EQUAL "3.7"))
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DUSE_PROTOBUF_CALLBACK_HEADER")
-endif()
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DUSE_PROTOBUF_CALLBACK_HEADER")
 
 #####################################################################
 # Enable code coverage calculation with gcov/gcovr/lcov
