From ed65f09c22b8395084332475cad8b4eb86e958bc Mon Sep 17 00:00:00 2001
From: Epo Jemba <epo.jemba@kametic.com>
Date: Thu, 12 Mar 2020 19:58:19 +0100
Subject: [PATCH] first fix for gnome 3.36

---
 extension.js  | 20 ++++++++++----------
 metadata.json |  4 +++-
 2 files changed, 13 insertions(+), 11 deletions(-)

diff --git a/extension.js b/extension.js
index b922e98..f345ff7 100644
--- a/extension.js
+++ b/extension.js
@@ -1460,7 +1460,7 @@ Grid.prototype = {
             reactive: true,
             width:this.tableWidth-20,
             height:36,
-            layout_manager: new Clutter.TableLayout()
+            layout_manager: new Clutter.GridLayout()
         });
         this.bottombar_table_layout = this.bottombar.layout_manager;
 
@@ -1479,7 +1479,7 @@ Grid.prototype = {
             reactive: true,
             width:this.tableWidth-20,
             height:36,
-            layout_manager: new Clutter.TableLayout()
+            layout_manager: new Clutter.GridLayout()
         });
         this.veryBottomBar_table_layout = this.veryBottomBar.layout_manager;
 
@@ -1498,7 +1498,7 @@ Grid.prototype = {
 
             let button = gridSettingsButton[index];
             button = new GridSettingsButton(button.text,button.cols,button.rows);
-            this.bottombar_table_layout.pack(button.actor, colNum, rowNum);
+            this.bottombar_table_layout.attach(button.actor, colNum, rowNum, 1, 1);
             button.actor.connect('notify::hover',Lang.bind(this,this._onSettingsButton));
             colNum++;
         }
@@ -1516,7 +1516,7 @@ Grid.prototype = {
             reactive: true,
             height:this.tableHeight,
             width:this.tableWidth-2,
-            layout_manager: new Clutter.TableLayout()
+            layout_manager: new Clutter.GridLayout()
         });
         this.table_table_layout = this.table.layout_manager;
         this.tableContainer.add_actor(this.table);
@@ -1541,19 +1541,19 @@ Grid.prototype = {
         }
 
         let toggle = new ToggleSettingsButton("animation",SETTINGS_ANIMATION);
-        this.veryBottomBar_table_layout.pack(toggle.actor, 0, 0);
+        this.veryBottomBar_table_layout.attach(toggle.actor, 0, 0, 1, 1);
         toggleSettingListener.addActor(toggle);
 
         toggle = new ToggleSettingsButton("auto-close",SETTINGS_AUTO_CLOSE);
-        this.veryBottomBar_table_layout.pack(toggle.actor, 1, 0);
+        this.veryBottomBar_table_layout.attach(toggle.actor, 1, 0, 1, 1);
         toggleSettingListener.addActor(toggle);
 
         let action = new AutoTileMainAndList(this);
-        this.veryBottomBar_table_layout.pack(action.actor, 2, 0);
+        this.veryBottomBar_table_layout.attach(action.actor, 2, 0, 1, 1);
         action.connect('resize-done', Lang.bind(this,this._onResize));
 
         action = new AutoTileTwoList(this);
-        this.veryBottomBar_table_layout.pack(action.actor, 3, 0);
+        this.veryBottomBar_table_layout.attach(action.actor, 3, 0, 1, 1);
         action.connect('resize-done', Lang.bind(this,this._onResize));
 
         this.x = 0;
@@ -1584,7 +1584,7 @@ Grid.prototype = {
 
                 this.elements[r][c] = element;
                 element.actor._delegate = this.elementsDelegate;
-                this.table_table_layout.pack(element.actor, c, r);
+                this.table_table_layout.attach(element.actor, c, r, 1, 1);
                 element.show();
             }
         }
@@ -1625,7 +1625,7 @@ Grid.prototype = {
         this.elementsDelegate.reset();
         let time = (gridSettings[SETTINGS_ANIMATION]) ? 0.3 : 0 ;
 
-        this.actor.raise_top();
+        Main.uiGroup.set_child_above_sibling(this.actor,null);
         Main.layoutManager.removeChrome(this.actor);
         Main.layoutManager.addChrome(this.actor);
         //this.actor.y = 0 ;
diff --git a/metadata.json b/metadata.json
index f260382..879e9cc 100644
--- a/metadata.json
+++ b/metadata.json
@@ -13,7 +13,9 @@
     "3.28", 
     "3.30", 
     "3.32",
-    "3.34"
+    "3.34",
+    "3.35",
+    "3.36"
   ], 
   "url": "https://github.com/gTile", 
   "uuid": "gTile@vibou", 
