diff --unified --recursive --text a/cli/justfile b/cli/justfile
--- a/cli/justfile	2025-12-21 21:18:14.000000000 -0700
+++ b/cli/justfile	2025-12-22 10:59:40.185222673 -0700
@@ -14,7 +14,6 @@
     cargo insta review
 
 install path:
-    cargo build --release
     ./scripts/install.sh "{{path}}"
 
 uninstall path:
diff --unified --recursive --text a/gui/justfile b/gui/justfile
--- a/gui/justfile	2025-12-21 21:18:14.000000000 -0700
+++ b/gui/justfile	2025-12-22 10:59:23.885435608 -0700
@@ -55,7 +55,6 @@
     cargo llvm-cov --no-report --no-cfg-coverage
 
 install path:
-    cargo build --release
     ./scripts/install.sh "{{path}}"
 
 uninstall path:
