diff --unified --recursive --text gnome-tour.orig/data/resources/style.css gnome-tour.new/data/resources/style.css
--- gnome-tour.orig/data/resources/style.css	2020-10-23 19:26:35.521089706 -0600
+++ gnome-tour.new/data/resources/style.css	2020-10-23 19:13:43.219486000 -0600
@@ -22,7 +22,7 @@
 }

 .last-page {
-  background-color: #4a86cf; /*GNOME blue*/
+  background-color: #3c8d7d; /*Manjaro teal*/
 }

 .last-page .page-title, .last-page .page-body {
diff --unified --recursive --text gnome-tour.orig/src/widgets/window.rs gnome-tour.new/src/widgets/window.rs
--- gnome-tour.orig/src/widgets/window.rs	2020-10-23 19:26:35.537757726 -0600
+++ gnome-tour.new/src/widgets/window.rs	2020-10-23 19:18:33.783489000 -0600
@@ -87,8 +87,8 @@
         self.paginator.borrow_mut().add_page(
             ImagePageWidget::new(
                 "/org/gnome/Tour/software.svg",
-                gettext("Use Software to find and install apps"),
-                gettext("Discover great apps through search, browsing and our recommendations."),
+                gettext("Use Add/Remove Software to find and install apps"),
+                gettext("Discover great apps through search, browsing and our recommendations in Manjaro Hello."),
             )
             .widget
             .upcast::<gtk::Widget>(),
@@ -97,7 +97,7 @@
         let name = glib::get_os_info("NAME").unwrap_or_else(|| "GNOME".into());
         let last_page = ImagePageWidget::new(
             "/org/gnome/Tour/ready-to-go.svg",
-            // Translators: The following string is formated as "We hope that you enjoy GNOME"
+            // Translators: The following string is formated as "We hope that you enjoy Manjaro Linux"
             i18n_f("That's it! We hope that you enjoy {}.", &[&name]),
             gettext("To get more advice and tips, see the Help app."),
         );
