--- main_window-a.py	2023-05-14 02:29:23.000000000 -0600
+++ main_window-b.py	2023-05-14 08:51:21.431324093 -0600
@@ -27,9 +27,9 @@
     CONFIG = f"{Path.home()}/.var/app/io.github.vikdevelop.SaveDesktop/config"
 else:
     try:
-        locale = open(f"translations/{r_lang}.json")
+        locale = open(f"/usr/share/savedesktop/translations/{r_lang}.json")
     except:
-        locale = open("translations/en.json")
+        locale = open("/usr/share/savedesktop/translations/en.json")
     os.system("mkdir ~/.config/io.github.vikdevelop.SaveDesktop")
     os.system("mkdir ~/.cache/io.github.vikdevelop.SaveDesktop")
     CACHE = f"{Path.home()}/.cache/io.github.vikdevelop.SaveDesktop"
