diff --unified --recursive --text a/justfile b/justfile
--- a/justfile	2025-06-15 12:23:35.553354020 -0600
+++ b/justfile	2025-06-15 12:24:23.509043609 -0600
@@ -179,8 +179,6 @@
     install -Dm0644 -t '{{DESTPREFIX}}/share/icons/hicolor/scalable/apps/' 'resources/icons/scalable/apps/{{APPID}}.svg'
     install -Dm0644 resources/icons/symbolic/apps/de.swsnr.turnon-symbolic.svg \
         '{{DESTPREFIX}}/share/icons/hicolor/symbolic/apps/{{APPID}}-symbolic.svg'
-    @# Compile settings schemas after installation
-    glib-compile-schemas --strict '{{DESTPREFIX}}/share/glib-2.0/schemas'
 
 # Assemble the README image from screenshots.
 build-social-image:
